dvc/output/base.py

```diff
@@ -309,12 +309,14 @@ def move(self, out):
         if self.scheme == "local" and self.use_cache and self.is_in_repo:
             self.repo.scm.ignore(self.path)
 
-    def get_files_number(self):
+    def get_cache_links_number(self):
         if not self.use_cache or not self.checksum:
             return 0
 
         if self.is_dir_checksum:
-            return len(self.dir_cache)
+            # NOTE double number of files because unpacked directory creates
+            # additional links
+            return len(self.dir_cache) * 2
```

Comment:
This is not obvious. Is `checkout` progress bar is using the same progress bar as on creating unpacked dir? And that is why you needed this change?

Reply:
Yes, should I make note more clear?

Reply:
@pared Just don't use the same progress bar :) Create a new progress bar for unpacking the dir and we won;t need these unintuitive multiplications